core/src/main/scala/fs2/job/Job.scala

```diff
@@ -17,6 +17,26 @@
 package fs2
 package job
 
+import scala.{Product, Serializable}
 import scala.util.Either
 
-final case class Job[F[_], I, N, R](id: I, run: Stream[F, Either[N, R]])
+import cats.Eq
+
+final case class Job[F[_], I, N, R](id: I, run: Stream[F, Either[N, R]], reportOn: N => Report)
```

Comment:
Instead of `reportOn`, how about a method `filterNotifications(N => Boolean): Job[F, I, N, R]` that applies the predicate to `run`. This makes it convenient to control notifications and doesn't require any special handling in the job manager.

Could also add a method like `def silent: Job[F, I, N, R] = filterNotifications(_ => false)`

If we wanted to make the `N` parameter scala-covariant we could also have `def silent: Job[F, I, Nothing, R] = ...`

Reply:
Oh that's very clever! Will implement.

Reply:
I agree that this would be a stronger design here.

Reply:
I'll implement this. Is it possible we might lose the effects attached to filtered notfications?

Reply:
Some testing indicates effects aren't lost. Added a test for it too.